eclair-core/src/main/scala/fr/acinq/eclair/db/Databases.scala

```diff
@@ -111,7 +113,19 @@ object Databases extends Logging {
           l.obtainExclusiveLock(ds)
           // ...and renew the lease regularly
           import system.dispatcher
-          system.scheduler.scheduleWithFixedDelay(l.leaseRenewInterval, l.leaseRenewInterval)(() => l.obtainExclusiveLock(ds))
+          val leaseLockTask = system.scheduler.scheduleWithFixedDelay(l.leaseRenewInterval, l.leaseRenewInterval)(() => l.obtainExclusiveLock(ds))
+
+          CoordinatedShutdown(system).addTask(CoordinatedShutdown.PhaseActorSystemTerminate, "release-postgres-lock") { () =>
+            Future {
+              logger.info("cancelling the pg lock renew task...")
+              leaseLockTask.cancel()
+              logger.info("releasing the curent pg lock...")
+              l.releaseExclusiveLock
```

Comment:
nit: it's quite weird that this is called like a pure function, whereas it has (strong) side-effects, shouldn't this be `l.releaseExclusiveLock(ds)`?

Reply:
Done with 61352e250.